tests/OmniSharp.Tests/UpdateBufferFilterFacts.cs

```diff
@@ -89,9 +89,18 @@ public UpdateBufferFilterFacts(ITestOutputHelper output)
                     filePath: "transient.cs");
 
                 var newSolution = host.Workspace.CurrentSolution.AddDocument(document);
+                host.Workspace.TryApplyChanges(newSolution);
 
+                // 2 transient + 1 real document.
                 docIds = host.Workspace.CurrentSolution.GetDocumentIdsWithFilePath("transient.cs");
-                Assert.Equal(2, docIds.Length);
+                Assert.Equal(3, docIds.Length);
+
+                // wait for event to be raised.
```

Comment:
@david-driscoll I pushed this version of test to illustrate what I think is going on (workspaceChange is async, test was checking too early). If we agree this is expected end result and find a more... elegant way of awaiting raising event, I will update test. I think previous test was not really testing much,

Reply:
I'd rather see a loop that awaits a `Task.Delay(...)` or a `Task.Yield(...)` and verifies that the event triggered rather than a single `Thread.Sleep(...)`.

Reply:
I think for now switching it from `Thread.Sleep` to `await Task.Delay(100)` is a better choice.  

You could also add a loop like with a cancellation token...

```csharp
var cts = new CancellationTokenSource();
cts.CancelAfter(TimeSpan.FromSeconds(10));
while (!cts.IsCancellationRequested) {
  await Task.Yield();
  docIds = host.Workspace.CurrentSolution.GetDocumentIdsWithFilePath("transient.cs");
  if (docIds.Count() <= 1) break;
}
Assert.Single(docIds);
```

It's a little (okay its a lot) janky, but it would ensure that test is a little more stable if it takes more than 100ms to update the workspace.